rust_dev_preview/iam/src/iam-service-lib.rs

```diff
@@ -39,6 +39,38 @@ pub async fn create_policy(
 }
 // snippet-end:[rust.example_code.iam.service.create_policy]
 
+#[cfg(test)]
+mod test_create_policy {
+    use crate::create_policy;
+    use sdk_examples_test_utils::single_shot_client;
+
+    #[tokio::test]
+    async fn test_create_policy_success() {
+        let client = single_shot_client!(
+            aws_sdk_iam,
+            "",
+            200,
+            include_str!("../testing/test_create_policy_response_success.xml")
+        );
```

Comment:
if you're writing a macro, I'd add field names here:
```
single_shot_client! {
  sdk: aws_sdk_iam,
  request (no idea what "" is here): "",
  status: 200,
  response: include_str!(".....")
}

Reply:
Oh nice